python/pyspark/sql/connect/client/artifact.py

```diff
@@ -243,11 +244,15 @@ def _create_requests(
         self, *path: str, pyfile: bool, archive: bool, file: bool
     ) -> Iterator[proto.AddArtifactsRequest]:
         """Separated for the testing purpose."""
-        return self._add_artifacts(
-            chain(
-                *(self._parse_artifacts(p, pyfile=pyfile, archive=archive, file=file) for p in path)
+        try:
+            yield from self._add_artifacts(
+                chain(
+                    *(self._parse_artifacts(p, pyfile=pyfile, archive=archive, file=file) for p in path)
+                )
             )
-        )
+        except Exception as e:
+            logger.error(f"Failed to submit addArtifacts request: {e}")
```

Comment:
I think we should show this error message regardless of the `SPARK_CONNECT_LOG_LEVEL` setting to make it more debuggable. Can we raise a more user-friendly error below?

Reply:
The problem is not that we don't raise error below, but that it will be intercepted by `grpc`, and it doesn't log what was thrown (I would argue that it definitely should, but.).

Reply:
In core.py, there is a method to handle errors from the server: `def _handle_error(error)`. It's able to handle various GRPC errors and also custom errors. Can we have something similar for add artifacts?

Reply:
Could you be more specific on what `Exception` can be thrown here? From your PR description, it looks like a GPRC error.

Reply:
What's the behavior after your change when the SPARK_CONNECT logger is enabled?

Reply:
I just think we shouldn't leave users this error message, when they don't enable the spark connect logger:
```
        status = StatusCode.UNKNOWN
        details = "Exception iterating requests!"
        debug_error_string = "None"
```

Reply:
In my example I'm intercepting FileNotFound error

Reply:
> What's the behavior after your change when the SPARK_CONNECT logger is enabled?
described in pr